### PR TITLE
Initial 1.3.0 commit

### PR DESCRIPTION
**Pull request by Yuri010**
# Main issues
 - [x] Path resolving when operating snapshots (currently gets lost in the bot_modules folder)
 - [x] Thorough testing to do
 - [x] Feedback messages apparently don't work well

# Todo
 - [x] Check paths
 - [x] Fix reply messages
 - [x] Test and debug
 - [x] Check snapshot name on creation if it doesn't exist already
 - [x] Perhaps make the embeds look a bit nicer and consolidate the creation process into one embed (Editing the original)
 - [x] Add a single line command to create snapshots instead of the whole process with prompts

Update aimed to make code easier to read :P

### CODE REVIEW

**Comment by Yuri010** [comment]
Tried adding these variables to modules that needed it, turns out that doesn't work;
```py
script_path = os.path.dirname(os.path.abspath(__file__))
root_path = os.path.join(script_path, '..')
db_path = os.path.join(root_path, 'minecraft_manager.db')
config_path = os.path.join(root_path, 'config.cfg')
```
Will have to look for a better way of resolving these paths...

Error details provided below.. Looks like it starts off just unable to find a path, and then throws a few errors about embeds that haven't even been defined yet, don't know what's up with that.

<details>
  <summary><i>Python Log</i></summary>
  
  ```py
  Traceback (most recent call last):
  File "C:\Users\YuriP\Downloads\MC-SRV\scripts\bot_modules\snapshots.py", line 77, in create_snapshot
    shutil.copytree(os.path.join(root_path, "..", folder), os.path.join(temp_folder, folder))
  File "C:\Users\YuriP\AppData\Local\Programs\Python\Python311\Lib\shutil.py", line 559, in copytree
    with os.scandir(src) as itr:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\YuriP\\Downloads\\MC-SRV\\scripts\\bot_modules\\..\\..\\world_nether'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\YuriP\Downloads\MC-SRV\scripts\bot_modules\snapshots.py", line 134, in create_snapshot
    description=f':x: Failed to create snapshot "{snapshot_name}": {str(e)}',
                                                  ^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'snapshot_name' where it is not associated with a value

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\YuriP\AppData\Local\Programs\Python\Python311\Lib\site-packages\discord\ext\commands\core.py", line 235, in wrapped
    ret = await coro(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\YuriP\Downloads\MC-SRV\scripts\bot.py", line 375, in snapshots_command
    await bot_modules.create_snapshot(ctx, bot, server_running, suppress_success_message=False)
  File "C:\Users\YuriP\Downloads\MC-SRV\scripts\bot_modules\snapshots.py", line 141, in create_snapshot
    await name_q.delete()
          ^^^^^^
UnboundLocalError: cannot access local variable 'name_q' where it is not associated with a value

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\YuriP\AppData\Local\Programs\Python\Python311\Lib\site-packages\discord\ext\commands\bot.py", line 1350, in invoke
    await ctx.command.invoke(ctx)
  File "C:\Users\YuriP\AppData\Local\Programs\Python\Python311\Lib\site-packages\discord\ext\commands\core.py", line 1029, in invoke
    await injected(*ctx.args, **ctx.kwargs)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\YuriP\AppData\Local\Programs\Python\Python311\Lib\site-packages\discord\ext\commands\core.py", line 244, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: UnboundLocalError: cannot access local variable 'name_q' where it is not associated with a value
  ```
  
</details>

**Comment by Yuri010** [comment]
Worth testing:
```py
# Get the absolute path of the script
script_path = os.path.dirname(os.path.abspath(__file__))

# Define the root directory (parent of the "scripts" folder)
root_path = os.path.abspath(os.path.join(script_path, '..'))
```
Instead of nesting "path.join" queries.

**Comment by Yuri010** [comment]
Alright.. That was fun.
Will try and resolve the remaining Pylint messages to make my code just perfect

Also still have to add the oneliner "create_snapshot" command and just thoroughly test everything. Ah well.\
Some other day :)

### Remaining Linting Issues

- [ ] **Global Statement**: The usage of the global statement should be reconsidered. Ideally, we can avoid globals by passing variables as parameters to functions.
 - [ ] **Resource Management**: Several places suggest using `with` statements for resource allocation, especially with file handling or socket connections, to ensure proper cleanup and management of resources.
- [ ] **Public IP Check**: There's an issue with checking if `public_ip` supports membership tests. Will have to ensure that `public_ip` is always a string before performing membership checks.
- [ ] **Complexity Warnings**: Some functions are flagged for having too many branches, local variables, return statements, or statements overall. Should consider refactoring to make them simpler, improving readability and maintainability.
- [ ] **Duplicate Code**: There's a similarity in code between two modules (`bot.py` and `verify.py`) that could be consolidated.

**Comment by Yuri010** [comment]
Could work on linting and optimisation a bit still.. But seems to be fine for now. Thorough testing up next

Updater really is a mess and could use revision.
Something for a future update

**Review by Yuri010 — COMMENTED** [review]
Or so I thought... Apparently pylint still returns there is duplicate code.. But seems so insignificant this won't be worked on.

**Comment by Yuri010** [comment]
I *think* that's it now?

Only stuff Pylint is whining about are complexity warnings.. But that's just how my code is. I could split everything up ofc but I don't know if that will make it any better.

Will have to update the ` updater.bat` so that works. Perhaps rework that whole thing as it's a mess currently.

But aside from that- I think this is basically ready to go?
Perhaps some long term testing?

### TL;DR To-Do
- [ ] Make sure updater works, perhaps rework.
- [ ] Testing, really making sure everything works

**Comment by Yuri010** [comment]
Updater dropped for now. Setting it up manually isn't really that big of a hassle..
Might fix eventually, but I have my doubts. Slash commands will be my main focus really.

For now just gonna test the bot thoroughly and push it ig
